### ISSUE TITLE
Alert when making datasets or visualizations public

### ISSUE THREAD

**Issue by danicarrion** [CartoDB/cartodb#13796]
When users publish a visualization that is publicly available, there must be a pop-up reminding them to make sure that they’re aware they also might be publishing personal data publicly. Suggested text: _You are about to publish this visualization for public view. Any personal data that you have used to create it will also be publicly available._

**Comment by ivanmalagon** [comment]
Let's use the same design as other warnings.

![image](https://user-images.githubusercontent.com/1078228/39051416-e1d33dcc-44a8-11e8-9b18-804804cddfed.png)

Possible issue: this will be a modal inside a modal (the one for publishing).

**Comment by jesusbotella** [comment]
Currently, we can edit maps privacy from builder and dashboard, and datasets privacy from dashboard and datasets page.

I think we have several behaviours that we need to sort out.

I am going to start by the builder ones.
In builder, we can change map's privacy from the publish modal, and the "privacy dropdown".

#### Publish modal
<img width="957" alt="screen shot 2018-04-23 at 12 57 42" src="https://user-images.githubusercontent.com/4319728/39122838-585c3d0a-46f6-11e8-9d2d-075ca53d6d14.png">

#### Privacy dropdown
<img width="320" alt="screen shot 2018-04-23 at 12 56 42" src="https://user-images.githubusercontent.com/4319728/39124263-18e90586-46fb-11e8-8a1d-0d446cb3a695.png">

I guess that we only need to show the alert when clicking the button to publish a map with the privacy set to public, because I have tested that the privacy is not being taken into account when the map is not published. But what happen when the map is published with private privacy and it is changed to public via dropdown, do we have to show the alert as well?

In dashboard, we have a privacy modal as well, do we need to take into account if the map is published to show the alert when the privacy is changed to public?

<img width="1071" alt="screen shot 2018-04-23 at 13 39 53" src="https://user-images.githubusercontent.com/4319728/39124459-db0a0264-46fb-11e8-90cc-a305d739cac1.png">

When it comes to datasets, there are no publish modal in the dataset view, only the privacy dropdown. So I think that we need to show the alert every time the privacy is changed to public, right?

We can change the datasets privacy in dashboard as well, so I think that we need to show the alert when changing the dataset privacy to public, right?

**Comment by ivanmalagon** [comment]
Good catch.

As you say, we cannot rely only on the publish button if the map is public as the ticket states. We need to show the notification everytime the map changes from whatever state it is to `published && public`. 

You're right also that we forgot about datasets. Setting a dataset to public shows it in the public user profile, making it accessible, so we must warn also when dataset privacy goes from `something` to `public`.

What do we consider as `public`? Accessible with link and password protected are also a kind of public since other people can read the data, even if you are controlling who they are somehow. Let's treat those states also as public, changing the warning copy. Are you ok with that @danicarrion @agnesgyorfi ??

**Comment by ivanmalagon** [comment]
Warning on updates would be overkilling. Only when the map goes to public.

**Comment by jesusbotella** [comment]
Thank you for your feedback @agnesgyorfi! I am going to answer my thoughts on that :)

1. We can change whichever text we want on the platform to make it clearer for users, and more GDPR<i>ish</i> 😁

2. We can change these texts, but we need to take into account that the default text inside the dropdown button is the current selected privacy option. So we will need another way to tell the user which one is the selected one outside the dropdown itself. Maybe design can help with this one as well.

3. Yes, the way I did this behaviour is: if the visualization is gonna be _public_, and the user clicks the publish button we warn the user by means of the alert. If it is not going to be _public_, and the user click the publish button we do nothing. But in the latter option, if the user changes the privacy via the dropdown we will warn the user by showing the alert because the map will be publicly available.
So, in other words, we only show the alert when the map is going to be publicly available.

The `Never published` text is an informative text. It shows that text if the map has never been published or the most recent map update date if the map is published.

I have another questions on top of those.
**Are we going to show the same text in the alert no matter which privacy the user selects?**

In dashboard, the behaviour is not consistent, because we only show the alert when the map is published and the privacy change is going to make the map publicly available. But we don't tell the user if the map has been already published. So, there's going to be an alert when making public some maps, and some others that won't show the alert. Is this correct?

**Comment by ivanmalagon** [comment]
shipped 🚀
